src/v2/Components/ViewInRoom/ViewInRoomTransition.tsx

```diff
@@ -0,0 +1,103 @@
+import { Box, Image } from "@artsy/palette"
+import React from "react"
+import { useLayoutEffect } from "react"
+import { useState } from "react"
+import { wait } from "./util"
+
+interface ViewInRoomTransitionProps {
+  children({ onMount }: { onMount(): void }): JSX.Element
+}
+
+export const ViewInRoomTransition: React.FC<ViewInRoomTransitionProps> = ({
+  children,
+}) => {
+  const [transitionState, setTransitionState] = useState<{
+    src?: string
+    srcSet?: string
+    from?: DOMRect
+    to?: DOMRect
+  }>({})
+
+  const [transitionStage, setTransitionStage] = useState<
+    "PENDING" | "TRANSITION" | "DONE"
+  >("PENDING")
+
+  const [isMounted, setMounted] = useState(false)
+
+  useLayoutEffect(() => {
+    const run = async () => {
+      const imgFrom = document.getElementById(
+        "transitionFrom--ViewInRoom"
+      ) as HTMLImageElement
+
+      const imgTo = document.getElementById(
+        "transitionTo--ViewInRoom"
+      ) as HTMLImageElement
+
+      setTransitionState({
+        src: imgFrom.src,
+        srcSet: imgFrom.srcset,
+        from: imgFrom.getBoundingClientRect(),
+        to: imgTo.getBoundingClientRect(),
+      })
+
+      await wait(100)
+      setTransitionStage("TRANSITION")
+
+      await wait(500)
```

Comment:
Since the 500 and 100 are used in a few places it might be worth moving them to a const at the top of the file

Reply:
I'm not sure how that'd improve the readability? It's pretty clear: wait for 100ms, wait for 500ms.

Reply:
Unless you're talking about interpolating them into the transition durations!

Reply:
I immediately wanted to re-write this with the [Animation API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Animations_API) but got bogged down in just making the transitions work... 🤔